<launch>
  <!-- Arguments -->

  <arg name="scan_topic"     default="scan"/>
  <!--<arg name="initial_pose_x" default="0.0"/>
  <arg name="initial_pose_y" default="0.0"/>
  <arg name="initial_pose_a" default="0.0"/>-->



  <!-- AMCL -->
  <node pkg="amcl" type="amcl" name="amcl">
    <rosparam command="load" file="/home/mecbot/Mecbot_ws/src/mecbot_driver_pkg/launch/odom_cat.yaml" />
    <param name="min_particles"             value="1000"/>  <!-- 500\1000 -->
    <param name="max_particles"             value="5000"/>  <!-- 5000 --> <!--2000 -->
    <param name="kld_err"                   value="0.01"/> <!-- 0.05\0.01 -->
    <param name="update_min_d"              value="0.1"/> <!-- 0.25 -->
    <param name="update_min_a"              value="0.20"/> <!-- 0.20 -->
    <param name="resample_interval"         value="2"/> <!-- 1/2 -->
    <param name="transform_tolerance"       value="0.1"/> <!-- 1/0.10 -->
    <param name="recovery_alpha_slow"       value="0.00"/>
    <param name="recovery_alpha_fast"       value="0.00"/>
   <!-- <param name="initial_pose_x"            value="$(arg initial_pose_x)" />
    <param name="initial_pose_y"            value="$(arg initial_pose_y)" />
    <param name="initial_pose_a"            value="$(arg initial_pose_a)" />-->
    <param name="gui_publish_rate"          value="10.0"/>

    <remap from="scan"                      to="$(arg scan_topic)"/>
    <param name="laser_max_range"           value="12"/>
    <param name="laser_max_beams"           value="180"/><!-- 180 --> <!-- 60 -->
    <param name="laser_z_hit"               value="0.5"/>
    <param name="laser_z_short"             value="0.05"/>
    <param name="laser_z_max"               value="0.05"/>
    <param name="laser_z_rand"              value="0.5"/>
    <param name="laser_sigma_hit"           value="0.2"/>
    <param name="laser_lambda_short"        value="0.1"/>
    <param name="laser_likelihood_max_dist" value="2.0"/>
    <param name="laser_model_type"          value="likelihood_field"/>

    <param name="odom_model_type"           value="omni-corrected"/>
    <param name="odom_alpha1"               value="0.2"/><!-- 0.2 -->
    <param name="odom_alpha2"               value="0.2"/><!-- 0.2 -->
    <param name="odom_alpha3"               value="0.2"/><!-- 0.2 -->
    <param name="odom_alpha4"               value="0.2"/><!-- 0.2 -->
    <param name="odom_alpha5"               value="0.1"/>
    <param name="global_frame_id"           value="map"/>
    <param name="odom_frame_id"             value="odom_cat"/>
    <param name="base_frame_id"             value="base_footprint"/>

  </node>
</launch>
